src/components/KMultiselect/KMultiselect.vue

```diff
@@ -0,0 +1,911 @@
+    // do the check off screen in the staging area so the UI doesn't jump
+    const stageSelections = () => {
+      // set timeout required to push the calculation to the end of the update lifecycle event queue
+      setTimeout(() => {
+        const elem = document.getElementById(multiselectSelectedItemsStagingId.value)
+
+        if (elem) {
+          const height = elem.clientHeight
+          if (height > selectionsMaxHeight.value) {
+            const item = visibleSelectedItemsStaging.value.pop()
+            if (item) {
+              invisibleSelectedItemsStaging.value.push(item)
+            }
+          }
+          stagingKey.value++
+        }
+      }, 0)
+    }
+
+    // handles programmatic selections
+    const handleMultipleItemsSelect = (items: MultiselectItem[]) => {
+      selectedItems.value = []
+      visibleSelectedItemsStaging.value = []
+      items.forEach(itemToSelect => {
+        const selectedItem = unfilteredItems.value.filter(anItem => anItem.key === itemToSelect.key)[0]
+        selectedItem.selected = true
+        selectedItem.key = selectedItem?.key?.includes('-selected') ? selectedItem.key : `${selectedItem.key}-selected`
+        selectedItem.key += '-selected'
+        selectedItems.value.push(selectedItem)
+        visibleSelectedItemsStaging.value.push(selectedItem)
+      })
+
+      stageSelections()
+    }
+
+    // handle item select/deselect from dropdown
+    const handleItemSelect = (item: MultiselectItem) => {
+      const selectedItem = unfilteredItems.value.filter(anItem => anItem.key === item.key)[0]
+      // if clicked item is already selected
+      if (selectedItem.selected) {
+        selectedItems.value = selectedItems.value.filter(anItem => anItem.key !== item.key)
+        // remove item from visibility arrays
+        if (visibleSelectedItemsStaging.value.filter(anItem => anItem.key === item.key).length) {
+          visibleSelectedItemsStaging.value = visibleSelectedItemsStaging.value.filter(anItem => anItem.key !== item.key)
+        } else if (invisibleSelectedItemsStaging.value.filter(anItem => anItem.key === item.key).length) {
+          invisibleSelectedItemsStaging.value = invisibleSelectedItemsStaging.value.filter(anItem => anItem.key !== item.key)
+        }
+        // deselect item
+        selectedItem.selected = false
+        selectedItem.key = selectedItem?.key?.replace(/-selected/gi, '')
+
+        // if some items are hidden grab the first hidden one and add it into the visible array
+        if (invisibleSelectedItemsStaging.value.length) {
+          const item = invisibleSelectedItemsStaging.value.pop()
+          if (item) {
+            visibleSelectedItemsStaging.value.push(item)
+          }
+        }
+      } else { // newly selected item
+        selectedItem.selected = true
+        selectedItem.key = selectedItem?.key?.includes('-selected') ? selectedItem.key : `${selectedItem.key}-selected`
+        selectedItem.key += '-selected'
```

Comment:
Something in the selected logic is flawed, as I can get selected items into a state that looks like this _(unsure if this is the right place to leave the comment)_

It looks like line `530` adds the `-selected` string in some cases, but then you're adding it again on the following line.

![image](https://user-images.githubusercontent.com/2229946/199349580-6ea75816-59b5-4458-909a-b5e1c69ea19e.png)